### ISSUE TITLE
Regression: Can not supply value files as URLs

### ISSUE THREAD

**Issue by davideagle** [argoproj/argo-cd#2762]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

Can not supply value files as URLs, they seem to be mistaken as file paths.

 helm template . --name oauth2-proxy --namespace ingress-nginx --kube-version 1.13 --values /tmp/https:__github.com_org_charts/bases/ingress-nginx/oauth2-proxy/

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

```yaml
applications:
  - name: oauth2-proxy
    project: ingress-nginx
    destination:
      server: https://kubernetes.default.svc
      namespace: ingress-nginx
    helm:
      chart: oauth2-proxy
      targetRevision: 2.0.0
      valueFiles:
        - https://raw.githubusercontent.com/org/Chart-values/master/oauth2-proxy/values-dev.yaml
    value_files:
      - values.yaml
    syncPolicy:
      automated:
        prune: true
        selfHeal: true
```

**Expected behavior**

Values should be loaded from URL

**Version**

```shell
argocd:v1.3.0-rc5
```

**Logs**

```
ComparisonError: rpc error: code = Unknown desc = `helm template . --name oauth2-proxy --namespace ingress-nginx --kube-version 1.13 --values /tmp/https:__github.com_org_charts/bases/ingress-nginx/oauth2-proxy/https:/raw.githubusercontent.com/Icelandair/Chart-values/master/oauth2-proxy/values-dev.yaml?` failed exit status 1: Error: open /tmp/https:__github.com_org_charts/bases/ingress-nginx/oauth2-proxy/https:/raw.githubusercontent.com/Icelandair/Chart-values/master/oauth2-proxy/values-dev.yaml no such file or directory
```

**Comment by alexec** [comment]
Interesting. Does `helm template --values` allow this?

**Comment by davideagle** [comment]
Yes it does.
 I suspect the problem lies with trying to treat a URL as a file path and prepending with /tmp/

 --values /tmp/https:__github.com_org_charts/bases/ingress-nginx/oauth2-proxy/

**Comment by alexec** [comment]
@jessesuen @alexmt  using HTTP URLs for values files is anti-GitOps, but I can see how it would be useful to Helm users, especially if they are using a  Helm chart rather than a Git repo. Thoughts?

**Comment by alexec** [comment]
Actually, this might be a regression? @simster7 FYI

**Comment by alexec** [comment]
@simster7 can I ask you to look into this? I thought we'd discussed not enforcing the directory traversal in `helm/cmd.go` as the purpose of this file is to dumbly invoke the command with the args passed to it. Instead this check should be the responsibility of the calling func (i.e. `helm.go#Template`.

See #2452

**Comment by alexec** [comment]
See #2716

**Comment by simster7** [comment]
Will look into this shortly

**Comment by simster7** [comment]
Another consideration regarding this is that allowing URL-based `values` files could essentially make the directory enforcer moot, since an attacker can always provide and host his own malicious `values` on the web and use them for an attack.

**Comment by simster7** [comment]
If we decide to allow URL-based `values` files and keep the directory enforcer, this issue could easily be fixed by moving the enforcer check higher up (as discussed in https://github.com/argoproj/argo-cd/pull/2716#discussion_r348870643) and providing an exception for URLs.

**Comment by alexec** [comment]
Thank you @simster7 . I'd like to wait for @jessesuen feedback, but I think it's likely we'll want to support this.

**Comment by simster7** [comment]
Come to think of it more... if an attacker had access to change the `ApplicationSourceHelm.ValueFiles` field to point to a malicious `values` file, why wouldn't the attacker simply inject the malicious values directly with `ApplicationSourceHelm.Values`?

This is convincing me more and more that the directory enforcer is superfluous.

**Comment by davideagle** [comment]
How about a separate values repository?

**Comment by alexec** [comment]
I think the enforcement check is still valid. We should not allow you to read from repositories you don't have access to.

**Comment by cbanek** [comment]
Just to chime in, we've seen this too, on 1.2.4 it seems to work, 1.3.0 it does not, with the same applications.

**Comment by simster7** [comment]
Yup, the changes that introduced this behavior are from 1.3.0

**Comment by simster7** [comment]
Closed with #2716
